server/desktop/src/main/java/dev/slimevr/desktop/tracking/trackers/hid/TrackersHID.kt

```diff
@@ -208,11 +207,17 @@ class TrackersHID(name: String, private val trackersConsumer: Consumer<Tracker>)
 						// tracker.batteryVoltage = if (battery and 128 == 128) 5f else 0f // Charge status
 						tracker.batteryVoltage = battery_mV.toFloat() * 0.001f
 						tracker.batteryLevel = (battery and 127).toFloat()
-						var rot = Quaternion(q[0], q[1], q[2], q[3])
-						rot = AXES_OFFSET.times(rot)
+						// The data comes in the same order as in the UDP protocol
+						// x y z w -> w x y z
+						var rot = Quaternion(q[3].toFloat(), q[0].toFloat(), q[1].toFloat(), q[2].toFloat())
```

Comment:
Existing HID devices sent rotation as WXYZ already, is it important to match UDP protocol here?

Reply:
It is not critical. However,
 - We change Q14 to Q15 anyway, so the firmware is subject to change.
 - XYZW has a minor performance advantage for SIMD optimizations compared to WXYZ
 - And most of the major APIs consider XYZW as standard:
   -  [.NET](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.quaternion.-ctor?view=net-8.0#system-numerics-quaternion-ctor(system-single-system-single-system-single-system-single))
   - [Unreal Engine](https://docs.unrealengine.com/4.27/en-US/API/Runtime/Core/Math/FQuat/)
   - [Apple](https://developer.apple.com/documentation/accelerate/working_with_quaternions)

Also, the unification with UDP will help to simplify firmware logic with #1001.

Reply:
Please note that documentation sometimes list properties/members in WXYW 
<img width="225" alt="image" src="https://github.com/SlimeVR/SlimeVR-Server/assets/845083/a2371b52-e3a8-404d-99fa-caa24a7f36b2">
because of alphabetical sorting.

What we need to look to is constructors:
<img width="520" alt="image" src="https://github.com/SlimeVR/SlimeVR-Server/assets/845083/fbaae852-c801-40e0-a181-e1e0a0e51829">

Reply:
> XYZW has a minor performance advantage for SIMD optimizations compared to WXYZ
any reference pls?

> And most of the major APIs consider XYZW as standard: 
yeah in APIs they use XYZW but in actual math its WXYZ

Reply:
> Also, the unification with UDP will help to simplify firmware logic with https://github.com/SlimeVR/SlimeVR-Server/pull/1001.

this doesnt matter because UDP logic and HID logic are separated, are we expecting in the future to somehow unify parsing logic?

Reply:
> any reference pls?

It is simple. Some operations need to work independently with XYZ and W.
For W, it doesn't matter where it is because it is a scalar. For XYZ, it matters in the `float4`lready; SIMD register a then it can de facto be used in `float3` SIMD library functions (if they are clever, they will use a single `float4` for most architectures instead of `float2` + `float`) because if it s. But, if the encoding is WXYZ, then the code needs to add a few assembler instructions to bring XYZ to the front of `float4`. That especially matters in heavily optimized operations with a lot of math because the number of SIMD registers may be relatively small for them. Compilers may fail to optimize the SIMD rotation of heavily optimized code, and that sometimes makes the code a few times slower, mostly because the amount of move operations exceeds the amount of floating-point operations.

> yeah in APIs they use XYZW but in actual math its WXYZ

Some math use WXYZ because of legacy code that was born without SIMD in mind.

> this doesnt matter because UDP logic and HID logic are separated, are we expecting in the future to somehow unify parsing logic?

The logic is unified in Gestures software. Although the fact that the sensors aren't slimes, I think it would be beneficial for both sides to make them work with NRF dongle and SlimeVR-Server.

Reply:
I don't have a link to some proof of my words. However, the following link help to ensure that I know what I am talking about https://github.com/libmir/mir-glas

Reply:
> Some operations

For example, a vector rotation.